cuppa/src/main/java/org/forgerock/cuppa/ConfigurationProvider.java

```diff
@@ -24,6 +26,8 @@
      * Should configure Cuppa using the provided {@link Configuration}.
      *
      * @param configuration The configuration that will be provided by Cuppa.
+     * @param properties : The per-provider specific properties that may come all the way from the plugin's
+     *      properties setting.
      */
-    void configure(Configuration configuration);
+    void configure(Configuration configuration, Map<String, String> properties);
```

Comment:
This appears to be a breaking change - is there anything we can do to avoid that?

Reply:
Looking through the changes, I can't actually see why this change is necessary for the described feature, anyway... Have I missed something?

Reply:
It's not necessary for the intersection group indeed. 
It's useful though if you want to register a TestTreeTransform that needs to be aware of the provider configuration.

I'm not keen to pass it in the Configuration because it's not the cuppa configuration. I can extract that change to another PR